src/lib.rs

```diff
@@ -503,6 +504,64 @@ impl KmerCountTable {
         Ok(())
     }
 
+    pub fn kmers_and_hashes(
+        &self,
+        seq: String,
+        skip_bad_kmers: bool,
+    ) -> PyResult<Vec<(String, u64)>> {
+        // TODO: optimize RC calculation
+        // TODO: confirm that there are no more hashes left? unreachable?
+        let seq = seq.to_ascii_uppercase();
+        let seqb = seq.as_bytes();
```

Comment:
Is the full contig/chromosome in memory twice at this point?

Reply:
no, it's a view (via the reference, `&`, see [as_bytes](https://doc.rust-lang.org/std/primitive.str.html#method.as_bytes)). 

If you calculate it, the memory consumption of the underlying string is minimal compared to catastrophic expansion of it into k-mers, even for human-size chromosomes.